engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -1635,6 +1677,24 @@ protected void runInContext() {
         }
     }
 
+    protected class CancelTask extends ManagedContextRunnable {
+        @Override
+        protected void runInContext() {
+            final List<AsyncJobVO> jobs = _asyncJobDao.getCancelledJobs();
```

Comment:
it might fetch again. but, it wont process again as they wont be in the sequence map.